File: Lab 05/tcp-udp-practice/TCPClient.py
from socket import *
serverName = "localhost"
serverPort = 12000
clientSocket = socket(AF_INET, SOCK_STREAM)
clientSocket.connect((serverName,serverPort))
sentence = bytes(input('Input lowercase sentence: '), 'utf_8')
clientSocket.send(sentence)
modifiedSentence = clientSocket.recv(1024)
print("From TCP Server:", modifiedSentence.decode('utf_8'))
clientSocket.close()

File: Lab 05/tcp-udp-practice/golang/TCPClient.go
package main

import (
	"fmt"
	"net"
)

func main() {
	serverPort := 12000
	serverAddress := &net.TCPAddr{
		IP:   net.ParseIP("localhost"),
		Port: serverPort,
	}
	serverSocket, err := net.DialTCP("tcp", nil, serverAddress)
	if err != nil {
		panic(err)
	}
	defer serverSocket.Close()

	fmt.Println("The client is ready to send")
	fmt.Printf("Enter message: ")
	var message string
	fmt.Scanln(&message)

	if _, err = serverSocket.Write([]byte(message)); err != nil {
		panic(err)
	}

	modifiedMessage := make([]byte, 2048)
	n, err := serverSocket.Read(modifiedMessage)
	fmt.Println("From TCP Server: ", string(modifiedMessage[:n]))
}
